doc: updated documentation for suppressionxpathfilter

Zopsss opened the pull request:
detected at: https://github.com/checkstyle/checkstyle/issues/15429#issuecomment-2293967608

corrected example queries

romani commented:
Please share CLI outputs to prove that we previous xpath did not work, and updated version is working.

romani commented:
You can use commit prefix `doc: ` in this case you update will be visible in release notes.

Zopsss commented:
GitHub, generate site

Zopsss commented:
> Please share CLI outputs to prove that we previous xpath did not work, and updated version is working.

code:

```java
import javax.annotation.processing.Generated;

/** some javadoc. */
public class TestingMethodName {
  @Generated("first") // should not be suppressed
  public void test_1() {}

  @Generated("second") // should be suppressed
  public void test_2() {}
}
```


suppression from doc:

```xml
    <module name="SuppressionXpathSingleFilter">
        <property name="checks" value="MethodName"/>
        <property name="query" value="//METHOD_DEF[.//ANNOTATION[
            ./IDENT[@text='Generated'] and ./EXPR/STRING_LITERAL[@text='second']]]"/>
        <property name="message" value=".*_.*"/>
    </module>
```

output:

```
Starting audit...
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestingMethodName.java
Starting audit...
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingMethodName.java:6:15: Method name 'test_1' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MethodName]
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingMethodName.java:9:15: Method name 'test_2' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MethodName]
Audit done.
```

fixed suppression:

```xml
    <module name="SuppressionXpathSingleFilter">
        <property name="checks" value="MethodName"/>
        <property name="query" value="//METHOD_DEF[.//ANNOTATION[
            ./IDENT[@text='Generated'] and ./EXPR/STRING_LITERAL[@text='second']]]//*"/>
        <property name="message" value=".*_.*"/>
    </module>
```

output:

```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestingMethodName.java
Starting audit...
[WARN] C:\checkstyle testing\TestingGoogleStyle\.\TestingMethodName.java:6:15: Method name 'test_1' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MethodName]
Audit done.
```

I have used google_checks.xml for whole config. The fixed suppression is working fine, the suppression provided at documentation isn't working at all. I tried tweaking the suppression but it wasn't working.

Zopsss commented on src/main/resources/google_checks.xml:
```diff
@@ -361,6 +355,12 @@
       <message key="name.invalidPattern"
              value="Method name ''{0}'' must match pattern ''{1}''."/>
     </module>
+    <module name="SuppressionXpathSingleFilter">
+      <property name="checks" value="MethodName"/>
+      <property name="query" value="//METHOD_DEF[
+                                     .//ANNOTATION/IDENT[@text='Test']]//*"/>
+      <property name="message" value="'[a-z][a-z0-9][_a-zA-Z0-9]*'.*"/>
+    </module>
```
Moved the suppression next to its module for easy reference, should have done this at: https://github.com/checkstyle/checkstyle/pull/15515/ but it slip through my mind so had to do it here

Zopsss replied:
Moved the suppression next to its module for easy reference, should have done this at: https://github.com/checkstyle/checkstyle/pull/15515 but it slip through my mind so had to do it here

romani reviewed (approved):
Ok to merge

Zopsss commented:
@rdiachenko

rdiachenko reviewed (approved):
lgtm
